main.js

```diff
@@ -24,3 +22,177 @@ form.addEventListener('submit', (event) => {
+  document.querySelectorAll('.normal-button').forEach((n) => n.addEventListener('click', () => {
+    modal.style.display = 'block';
+    modal.innerHTML = ` <div class="popup-card">
+<div class = "popup-close">
+        <h1 class = "popup-h1">${project.title}</h1>
+        
+</div>
+        <div class = "popup-frame">
+        <ul class="popup-frame">
+          <li class="detail-1">CANOPY</li>
+          <li class="detail">BACK END DEV</li>
+          <li class="detail">2015</li>
+        </ul>
+        </div>
+        <div class = "popup-img-card" style = "background-image: url('.${project.image}'); max-height: 300px; "></div>
+<div class = "popup-description">
+<div class = "desc">Lorem Ipsum is simply dummy text of the printing and typesetting industry.
+Lorem Ipsum has been the industry's standard dummy text ever since the 1500s,
+when an unknown printer took a galley of type and scrambled it to make a type specimen book.
+It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.
+It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, </div>
+<div class = "pop-frame">  
+ <div class="tags">
+<button class="tag">html</button>
+<button class="tag">css</button>
+<button class="tag">javascript</button>
+</div>
+<div class="tags popup-tag">
+<button class="tag">github</button>
+<button class="tag">ruby</button>
+<button class="tag">Bootstraps</button>
+</div>
+<hr class = "popup-hr">
+<div class = "popup-link-button">
+<button class = "popup-button">See More <img src = "./image/pop.svg"></button>
+<button class = "popup-button">See More <img src = "./image/Icon-GitHub.svg"></button>
+</div>
+</div>
+      </div> `;
+  }));
+});
+
+window.onclick = function (event) {
+  if (event.target === modal) {
+    modal.style.display = 'none';
+  }
+};
+
+const inputFields = document.querySelectorAll('input');
+inputFields.forEach((input) => {
+  input.addEventListener('change', (event) => {
+    let formData = JSON.parse(localStorage.getItem('formData'));
+    if (!formData) {
+      formData = { name: '', email: '', message: '' };
+    }
+    formData[event.target.name] = event.target.value;
+    localStorage.setItem('formData', JSON.stringify(formData));
+  });
+});
+
+const textArea = document.getElementById('message');
+textArea.addEventListener('change', (event) => {
+  let formData = JSON.parse(localStorage.getItem('formData'));
+  if (!formData) {
+    formData = { name: '', email: '', message: '' };
+  }
+  formData.message = event.target.value;
+  localStorage.setItem('formData', JSON.stringify(formData));
+});
+
+function retrieveFormData() {
+  const formData = JSON.parse(localStorage.getItem('formData'));
+  if (formData) {
+    document.getElementById('name').value = formData.name;
+    document.getElementById('email').value = formData.email;
+    document.getElementById('message').value = formData.message;
+  }
+}
+
```

Comment:
**[REQUIRED CHANGES]**

- [ ] hey, great job on preserving your data but there is an issue with the preservation.
- [ ] when a user fills up your form without submitting it after reloading the page, the email input and the name input get saved but the message input doesn't get saved. it only gets saved after submitting the form. please kindly ensure all your form data gets saved before submitting the form.

Reply:
**[REQUIRED CHANGES]**

- [ ] hey, great job on preserving your data but there is an issue with the preservation.
- [ ] when a user fills up your form without submitting it after reloading the page, the email input and the name input get saved but the message input doesn't get saved. it only gets saved after submitting the form. please kindly ensure all your form data gets saved before submitting the form.